src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,245 @@
+package duke;
+
+import duke.command.AddTaskCommand;
+import duke.command.Command;
+import duke.command.DeleteCommand;
+import duke.command.DoneCommand;
+import duke.command.EmptyInputCommand;
+import duke.command.ExitCommand;
+import duke.command.FindCommand;
+import duke.command.ListCommand;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Todo;
+
+import java.time.DateTimeException;
+import java.util.HashMap;
+
+/**
+ * Deals with making sense of user commands.
+ */
+public class Parser {
+    protected static HashMap<String, String> commandDelimiter;
+    protected static HashMap<String, String> commandTypeFormatInfo;
+
+    /**
+     * Creates a new command based on given input string.
+     *
+     * @param fullCommand represents a user input command
+     * @return Command based on action to be performed
+     * @throws DukeException when given input is not a valid command
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        setupCommandDelimiter();
+        setupCommandTypeFormatInfo();
+        // Remove leading and trailing whitespace
+        fullCommand = fullCommand.trim();
+        String[] commandWords = fullCommand.split("\\s+");
+        int numberOfCommandArguments = commandWords.length - 1;
+        Command outputCommand;
```

Comment:
I think it's better not to leave this outputCommand uninitialized, by returning a new command in ever case line.